### ISSUE TITLE
Webui user favorites will not accept entry without a folder.

### ISSUE THREAD

**Issue by mckayERP** [adempiere/adempiere#2324]
The current operation is not intuitive.  Needs a default folder or the ability to have menu items at the root level.  Also needs a good tooltip to explain it's use.

**Comment by marcalwestf** [comment]
@mckayERP nice improvement Mike. 
Users don't find it intuitive ans this is a pity because it is a helpful functionality!

**Comment by yamelsenih** [comment]
Hi @mckayERP here a error with Ids over 1000000
![Screenshot_20190315_151515](https://user-images.githubusercontent.com/2333092/54456882-4b618880-4736-11e9-989f-fc57e365beee.png)

<pre>
===========> Trx.commit: POSave_eac93056-144a-4caa-8539-30d1e4ad73d9 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_555b380d-d347-429b-8512-74689d73745d [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_0b034f39-7545-4a64-b26b-e413f35a0e55 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_4b288e4c-2fcf-47be-a7ac-ec056bea5888 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_8daf6363-9aac-4dae-86a2-5d101d8c3b4d [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_dea68c6d-1037-422d-91a7-4f87323f2f83 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_2e1aa8d0-fe3a-459c-b41a-dc852f85ec41 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_fe32e8bc-0764-4846-92c0-c976726555f6 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_ccc48f8d-1ef6-4bf8-bcb0-50b27d0d1006 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_5ce6b160-560e-46a5-afe4-b9ac951c7df6 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_91f8d7b5-54ed-40a3-a506-d32a2fd79425 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_da19ec41-a2ee-430b-8376-e7044240edad [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_ba0e166c-662d-42d6-acf0-d7a1667e6ae9 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_d3a02c8a-1256-40d7-ace8-7d8df9f79ab7 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_2e9dce62-c1aa-4dab-b2cd-353853648619 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_50a295a5-cf17-43a8-8de8-acd0f466fd27 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_e8e191c3-8e04-44f2-8772-fa33d9d22680 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_d8029f03-ef75-45ae-8114-b44186d37e19 [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)

===========> Trx.commit: POSave_28df26a8-d8a9-4e06-aea1-806a23b28e2d [41]
org.postgresql.util.PSQLException: ERROR: insert or update on table "ad_tree_favorite_node" violates foreign key constraint "adtreefavorite_adtreefavoriten"
  Detail: Key (ad_tree_favorite_id)=(1000000) is not present in table "ad_tree_favorite".; State=23503; ErrorCode=0
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgConnection.executeTransactionCommand(PgConnection.java:755)
	at org.postgresql.jdbc.PgConnection.commit(PgConnection.java:777)
	at com.mchange.v2.c3p0.impl.NewProxyConnection.commit(NewProxyConnection.java:803)
	at org.compiere.util.Trx.commit(Trx.java:336)
	at org.compiere.util.Trx.commit(Trx.java:364)
	at org.compiere.model.PO.save(PO.java:2198)
	at org.compiere.model.PO.saveEx(PO.java:2278)
	at org.compiere.model.PO.saveEx(PO.java:2406)
	at org.compiere.model.MMigrationStep.applyPO(MMigrationStep.java:602)
	at org.compiere.model.MMigrationStep.apply(MMigrationStep.java:229)
	at org.compiere.model.MMigration.apply(MMigration.java:151)
	at org.compiere.process.MigrationApply.doIt(MigrationApply.java:53)
	at org.compiere.process.SvrProcess.process(SvrProcess.java:175)
	at org.compiere.process.SvrProcess.startProcess(SvrProcess.java:128)


</pre>

**Comment by mckayERP** [comment]
Hi Yamel,

Sorry, should have put a note here to mention that.  I'll update the xml.

Seems there is a bug when I'm running eclipse with both the web and client at the same times.  I was using the client as System to make changes to translations and the webui as SuperUser to test them.  The things I changed as SuperUser in the GardenAdmin role were picked up in the client migration and I didn't notice until I did a new import of the seed.

Mike.
